ratpack-cassandra-tracing/src/main/java/smartthings/ratpack/cassandra/zipkin/TracedSession.java

```diff
@@ -0,0 +1,31 @@
+package smartthings.ratpack.cassandra.zipkin;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+import com.github.kristofa.brave.Brave;
+import com.google.inject.Inject;
+import smartthings.ratpack.cassandra.AbstractSession;
+
+public final class TracedSession extends AbstractSession {
+
+	private final Brave brave;
+	private final Cluster cluster;
+	private final String keyspace;
+	private final CassandraTracingModule.Config config;
+
+	@Inject
+	public TracedSession(Cluster cluster, Brave brave, CassandraTracingModule.Config config) {
+		super(cluster, config.getKeyspace());
+		this.brave = brave;
+		this.cluster = cluster;
+		this.keyspace = config.getKeyspace();
+		this.config = config;
+	}
+
+	protected Session createDelegate() {
+		Session session = (keyspace != null && !keyspace.equals("")) ? cluster.connect(keyspace) : cluster.connect();
+
+		return smartthings.brave.cassandra.TracedSession.create(session, brave, config.getServiceName());
```

Comment:
Here's the stacktrace Brian is seeing:

```
SEVERE: RuntimeException while executing runnable smartthings.brave.cassandra.TracedSession$TracedResultSetFuture$$Lambda$1@30335d5c with executor MoreExecutors.directExecutor()
ratpack.exec.UnmanagedThreadException: Operation attempted on non Ratpack managed thread 'cluster1-nio-worker-2'
    at ratpack.exec.internal.DefaultExecution.require(DefaultExecution.java:104)
    at ratpack.exec.Execution.current(Execution.java:81)
    at ratpack.zipkin.internal.RatpackServerClientLocalSpanState.setCurrentServerSpan(RatpackServerClientLocalSpanState.java:104)
    at com.github.kristofa.brave.ServerSpanThreadBinder.setCurrentSpan(ServerSpanThreadBinder.java:52)
    at smartthings.brave.cassandra.TracedSession$TracedResultSetFuture.lambda$addListener$0(TracedSession.java:404)
    at smartthings.brave.cassandra.TracedSession$TracedResultSetFuture$$Lambda$1.run(Unknown Source)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:456)
    at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:817)
    at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:753)
    at com.google.common.util.concurrent.AbstractFuture.set(AbstractFuture.java:613)
    at com.datastax.driver.core.DefaultResultSetFuture.onSet(DefaultResultSetFuture.java:174)
    at com.datastax.driver.core.RequestHandler.setFinalResult(RequestHandler.java:174)
    at com.datastax.driver.core.RequestHandler.access$2600(RequestHandler.java:43)
    at com.datastax.driver.core.RequestHandler$SpeculativeExecution.setFinalResult(RequestHandler.java:793)
    at com.datastax.driver.core.RequestHandler$SpeculativeExecution.onSet(RequestHandler.java:496)
    at com.datastax.driver.core.Connection$Dispatcher.channelRead0(Connection.java:1012)
    at com.datastax.driver.core.Connection$Dispatcher.channelRead0(Connection.java:935)
```

From a really quick look, my guess is that the traced version has brave internals operating in a non-ratpack way.  I think the problem might be here:

https://github.com/SmartThingsOSS/ratpack-cassandra/blob/master/ratpack-cassandra/src/main/java/smartthings/ratpack/cassandra/CassandraService.java#L105

I wonder if changing to a Blocking.get would solve for both (Traced and Non-Traced).